File: CQUPTSchedule/Utils/Extensions.swift
import SwiftUI


extension Color {
    // 接受当前索引和总课程数来均分色环
    static func dynamicCourseColor(index: Int, total: Int) -> Color {
        guard total > 0 else { return .blue }
        
        // 使用质数步长跳跃选色，避免色环上相邻的颜色太像
        let step = 7
        let steppedIndex = (index * step) % total
        let hue = Double(steppedIndex) / Double(total)
        
        // 亮度 0.6 确保白色文字清晰
        return Color(hue: hue, saturation: 0.7, brightness: 0.6)
    }
    
    static func dynamicCourseColor(index: Int) -> Color {
        let goldenRatio = 0.618033988749895
        let hue = (Double(index) * goldenRatio).truncatingRemainder(dividingBy: 1.0)
        return Color(hue: hue, saturation: 0.65, brightness: 0.75)
    }
}

extension Date {
    func formatToSchedule() -> String {
        let f = DateFormatter(); f.dateFormat = "yyyy/M/d"; return f.string(from: self)
    }
}

File: lib/utils/extensions.dart
import 'package:flutter/material.dart';
import 'package:intl/intl.dart';

class ColorExtensions {
  static Color dynamicCourseColor({required int index, required int total}) {
    if (total <= 0) return Colors.blue;
    
    final safeIndex = index.abs() % total;
    const step = 7;
    final steppedIndex = (safeIndex * step) % total;
    final hue = steppedIndex / total;
    
    return HSVColor.fromAHSV(1.0, hue * 360, 0.7, 0.6).toColor();
  }

  static Color dynamicCourseColorSimple(int index) {
    const goldenRatio = 0.618033988749895;
    final hue = (index * goldenRatio) % 1.0;
    return HSVColor.fromAHSV(1.0, hue * 360, 0.65, 0.75).toColor();
  }

  static List<Color> generateHSLColorPalette(int count) {
    if (count <= 0) return [];
    
    final colors = <Color>[];
    for (int i = 0; i < count; i++) {
      final hue = i / count;
      colors.add(HSVColor.fromAHSV(1.0, hue * 360, 0.7, 0.6).toColor());
    }
    return colors;
  }
}

extension DateTimeExtension on DateTime {
  String formatToSchedule() {
    return DateFormat('yyyy/M/d').format(this);
  }
}
